canonical-web-and-design/maas-ui
# feat(ui): add ability to create RAID 0 storage devices

## Done

- Added ability to create RAID 0 devices. The rest of the RAID levels and the storage devices table will come in a later PR.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a Ready or Allocated machine
- Create two unformatted partitions
- Select them both and check that the "Create RAID" button becomes enabled
- Click "Create RAID" and change some of the values
- Check that a RAID 0 device is created with the values you entered

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2166
